web/src/engine/websites/ManhwaHentaiMe.ts

```diff
@@ -1,21 +1,70 @@
-import { Tags } from '../Tags';
+import { type Tag, Tags } from '../Tags';
 import icon from './ManhwaHentaiMe.webp';
-import { DecoratableMangaScraper } from '../providers/MangaPlugin';
+import { Chapter, DecoratableMangaScraper, Manga, type MangaPlugin } from '../providers/MangaPlugin';
 import * as Madara from './decorators/WordPressMadara';
 import * as Common from './decorators/Common';
+import { FetchCSS, FetchHTML } from '../platform/FetchProvider';
 
-@Madara.MangaCSS(/^{origin}\/webtoon\/[^/]+\/$/, 'div.post-title h1')
-@Madara.MangasMultiPageAJAX()
-@Madara.ChaptersSinglePageAJAXv2()
+type MangaID = {
+     post: string;
+     slug: string;
+}
+
+function MangaInfoExtractor(anchor: HTMLAnchorElement) {
+    const container = anchor.closest<HTMLElement>('div.page-item-detail, div.manga');
+    const post = container?.querySelector<HTMLElement>('div[id*="manga-item-"]')?.getAttribute('id').match(/(\d+$)/)[1] || '';
+    const slug = anchor.pathname;
+    const title = anchor.text.replace(/Webtoon Manhwa Hentai$/, '').trim();
+    const id = JSON.stringify({ post, slug });
+    return { id, title };
+}
+
+@Common.MangasMultiPageCSS('/home/page/{page}/', 'div.post-title h3 a, div.post-title h5 a', 1, 1, 0, MangaInfoExtractor)
 @Madara.PagesSinglePageCSS()
 @Common.ImageAjax()
 export default class extends DecoratableMangaScraper {
 
-    public constructor() {
-        super('manhwahentaime', 'ManhwaHentai.me', 'https://manhwahentai.me', Tags.Media.Manhua, Tags.Media.Manhwa, Tags.Language.English);
+    public constructor(id: string = 'manhwahentaime', label: string = 'ManhwaHentai.me', url: string = 'https://manhwahentai.me', tags: Tag[] = [Tags.Media.Manhua, Tags.Media.Manhwa, Tags.Language.English]) {
+        super(id, label, url, ...tags);
     }
 
     public override get Icon() {
         return icon;
     }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return new RegExpSafe(`^${this.URI.origin}/webtoon/[^/]+\/$`).test(url);
+    }
+
+    public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
+        const uri = new URL(url);
+        const data = await FetchHTML(new Request(uri));
+        const post = data.querySelector<HTMLElement>('div#star[data-id]')?.dataset?.id;
+        const slug = uri.pathname;
+        const element = data.querySelector<HTMLElement>('div.post-title h1');
+        const title = Common.AnchorInfoExtractor(false, 'span').call(this, element).title.replace(/Webtoon Manhwa Hentai$/, '').trim();
+        return new Manga(this, provider, JSON.stringify({ post, slug }), title);
+    }
+
+    public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
```

Comment:
Is this something specific to this website, or is this something generic that may be adapted by other Madara websites as well?
In the latter case it would make sense to implement this method as decorator.

Reply:
I dont think 2 websites is enough to make a specific decorator. 
On the other hand

* we can always add the postid element finding in MangaCSS (since we already have a couple of them)
* A JS decorator to get chapters from Madara Manga IDentifier  ( post + slug) can always be handy. We never know who can block admin-ajax with Cloudflare